src/components/HeaderForm.js

```diff
@@ -172,6 +172,7 @@ class HeaderForm extends Component {
   componentDidMount() {
     this.fetchCanceler = new AbortController();
     this.cancelSignal = this.fetchCanceler.signal;
+    this.api = new APIInterface(this.props.apiUrl, this.cancelSignal);
```

Comment:
It looks like now we're making a fresh `APIInterface` in every React component that needs to talk to the API, and the components still take a string `apiUrl` as a prop.

This lays some of the groundwork we need, but when we come up with another implementation of `APIInterface` that isn't URL-based (`APIInterface2` or whatever), how are we going to plug it into the components?

I think what we should do is make the components all take the `APIInterface` instance itself as a prop, instead of taking a URL and making an `APIInterface` in each component. Then we can have one `APIInterface` for the whole application, and we can control where to get the data based on what one we send in from the top.